tests/src/JIT/Regression/CLR-x86-JIT/V1.2-Beta1/b103058/b103058.cs

```diff
@@ -31,17 +32,43 @@ private static unsafe bool CheckDoubleAlignment1(VT* p)
         }
     }
 
+    private static unsafe bool CheckAlignment1(VT* p)
+    {
+        Console.WriteLine("Address {0}", (IntPtr)p);
+        if ((int)(long)p % 4 != 0)
+        {
+            Console.WriteLine("not 4-byte aligned");
+            return false;
+        }
+        else
+        {
+            return true;
+        }
+    }
+
     public static int Main()
     {
         VT vt1 = new VT();
         VT vt2 = new VT();
         bool retVal;
 
-        retVal = CheckDoubleAlignment1(&vt1);
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || (RuntimeInformation.ProcessArchitecture != Architecture.X86))
+        {
+            retVal = CheckDoubleAlignment1(&vt1);
+
+            VT vt3 = new VT();
+            retVal = CheckDoubleAlignment1(&vt2) && retVal;
+            retVal = CheckDoubleAlignment1(&vt3) && retVal;
+        }
+        else
+        {
+            // The System V ABI for i386 defines double as having 4-byte alignment
```

Comment:
This test is testing performance optimization in the JIT that tries to get doubles aligned on stack. The JIT can do this performance optimization even for System V ABI in this case.

The alignment of doubles used to matter a lot for performance. It probably does not matter as much on current hardware. I think it is ok to not have this optimization for Linux x86. The comments should be fixed to match.

Reply:
done